rosetta/converter/converter.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package converter
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/encoding/json"
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/optakt/flow-dps/rosetta/object"
+)
+
+type Converter struct {
+	deposit    flow.EventType
+	withdrawal flow.EventType
+}
+
+func New(gen Generator) (*Converter, error) {
+	deposit, err := gen.TokensDeposited(dps.FlowSymbol)
+	if err != nil {
+		return nil, fmt.Errorf("unable to generate deposit event type: %w", err)
+	}
+	withdrawal, err := gen.TokensWithdrawn(dps.FlowSymbol)
+	if err != nil {
+		return nil, fmt.Errorf("unable to generate withdrawal event type: %w", err)
+	}
+
+	c := Converter{
+		deposit:    flow.EventType(deposit),
+		withdrawal: flow.EventType(withdrawal),
+	}
+
+	return &c, err
+}
+
+func (c *Converter) EventToOperation(event flow.Event) (operation *object.Operation, isRelevant bool, err error) {
+	switch event.Type {
+	case c.deposit, c.withdrawal:
+		isRelevant = true // Supported types, event is relevant for us.
+	default:
+		return nil, false, nil // This type of event is irrelevant for us.
+	}
+
+	// Decode the event payload into a Cadence value and cast it to a Cadence event.
+	value, err := json.Decode(event.Payload)
+	if err != nil {
+		return nil, isRelevant, fmt.Errorf("could not decode event: %w", err)
+	}
+	e, ok := value.(cadence.Event)
+	if !ok {
+		return nil, isRelevant, fmt.Errorf("could not cast event: %w", err)
```

Comment:
You are returning the explicit `isRelevant` here every time. I would just like to insist on the comment I made on another PR, that return values _should never be used_ when there is an error returned from a function. There are exceptions, even in the standard library, with sentinel errors. Beyond that, though, it's recommended not to rely on a return value's accuracy when an error is returned (and I think this makes sense).

In this case, there is really no need to inform anyone that the event is relevant for an operation, if it can't be converted to the relevant operation. The boolean return parameter should really just be used to indicate that no operation was produced, but that this is not an error, but in fact expected.